api/src/main/java/org/cardanofoundation/rosetta/api/block/service/LedgerBlockServiceImpl.java

```diff
@@ -131,77 +162,117 @@ public BlockIdentifierExtended findGenesisBlockIdentifier() {
             .map(transactionMapper::mapStakeRegistrationEntityToStakeRegistration)
             .toList());
 
     transaction.setDelegations(
-        delegationRepository
-            .findByTxHash(transaction.getHash())
+        fetched.delegations
             .stream()
+            .filter(tx -> tx.getTxHash().equals(transaction.getHash()))
             .map(transactionMapper::mapDelegationEntityToDelegation)
             .toList());
 
-    transaction.setPoolRegistrations(poolRegistrationRepository
-        .findByTxHash(transaction.getHash())
-        .stream()
-        .map(transactionMapper::mapEntityToPoolRegistration)
-        .toList());
+    transaction.setWithdrawals(
+        fetched.withdrawals
+            .stream()
+            .filter(tx -> tx.getTxHash().equals(transaction.getHash()))
+            .map(transactionMapper::mapWithdrawalEntityToWithdrawal)
+            .toList());
 
-    transaction.setPoolRetirements(poolRetirementRepository
-        .findByTxHash(transaction.getHash())
-        .stream()
-        .map(transactionMapper::mapEntityToPoolRetirement)
-        .toList());
+    transaction.setPoolRegistrations(
+        fetched.poolRegistrations
+            .stream()
+            .map(transactionMapper::mapEntityToPoolRegistration)
+            .toList());
 
-    transaction.setWithdrawals(withdrawalRepository
-        .findByTxHash(transaction.getHash())
-        .stream()
-        .map(transactionMapper::mapWithdrawalEntityToWithdrawal)
-        .toList());
+    transaction.setPoolRetirements(
+        fetched.poolRetirements
+            .stream()
+            .map(transactionMapper::mapEntityToPoolRetirement)
+            .toList());
 
-    ProtocolParams pps = protocolParamService.findProtocolParametersFromIndexer();
     transaction.setSize(calcSize(transaction, pps));
   }
 
   private static long calcSize(BlockTx tx, ProtocolParams p) {
     return (Long.parseLong(tx.getFee()) - p.getMinFeeB().longValue()) / p.getMinFeeA().longValue();
   }
 
-  private void populateUtxo(Utxo utxo) {
-    AddressUtxoEntity first = addressUtxoRepository
-        .findAddressUtxoEntitiesByOutputIndexAndTxHash(utxo.getOutputIndex(), utxo.getTxHash())
-        .getFirst();
-    Optional.ofNullable(first).ifPresent(m -> addressUtxoEntityToUtxo.overWriteDto(utxo,m));
+  private void populateUtxo(Utxo utxo, Map<UtxoKey, AddressUtxoEntity> utxoMap) {
+    AddressUtxoEntity entity = utxoMap.get(
+        new UtxoKey(utxo.getTxHash(), utxo.getOutputIndex()));
+    Optional.ofNullable(entity)
+        .ifPresent(e -> addressUtxoEntityToUtxo.overWriteDto(utxo, e));
+  }
+
+  private static Map<UtxoKey, AddressUtxoEntity> getUtxoMapFromEntities(Entities fetched) {
+    return fetched.utxos
+        .stream()
+        .collect(Collectors.toMap(
+            utxo -> new UtxoKey(utxo.getTxHash(), utxo.getOutputIndex()),
+            utxo -> utxo
+        ));
+  }
+
+  private record Entities(List<AddressUtxoEntity> utxos,
```

Comment:
They are all connected to a transaction. Let's rename it to `TransactionDetails`.

Reply:
Fixed